controllers/openshift_controller.go

```diff
@@ -850,72 +850,74 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
+		r.Log.Info("Waiting for MCO to start updating.")
+		// We don't requeue, an MCP going Updated->Updating will
+		// trigger reconciliation by itself thanks to our watching MCPs.
+		return reconcile.Result{}, nil
+	} else {
+		r.Log.Info("No need to wait for MCO to start updating.", "isMcoUpdating", isMcoUpdating, "Status.WaitingForMcoToStart", r.kataConfig.Status.WaitingForMcoToStart)
+		r.kataConfig.Status.WaitingForMcoToStart = false
+	}
+
 	r.clearUninstallStatus()
 	_, result, err2, done := r.updateStatus(machinePool)
 	if !done {
 		return result, err2
 	}
+	err = r.Client.Status().Update(context.TODO(), r.kataConfig)
+	if err != nil {
+		r.Log.Error(err, "Unable to update KataConfig status")
+		return ctrl.Result{}, err
+	}
 
-	if mcp.Status.ReadyMachineCount != mcp.Status.MachineCount {
-		return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+	if isMcoUpdating {
+		r.Log.Info("Waiting for MachineConfigPool to be fully updated", "machinePool", targetPool)
+		return reconcile.Result{}, nil
 	}
 
-	// Sleep for MCP to reflect the changes
-	r.Log.Info("Pausing for a minute to make sure mcp has started syncing up")
-	time.Sleep(60 * time.Second)
+	r.kataConfig.Status.UnInstallationStatus.InProgress.IsInProgress = corev1.ConditionFalse
 
-	//This is not applicable for converged cluster
-	isConvergedCluster, _ := r.checkConvergedCluster()
 	if !isConvergedCluster {
-		//Get "worker" MCP
-		wMcp := &mcfgv1.MachineConfigPool{}
-		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: "worker"}, wMcp)
-		if err != nil {
-			r.Log.Error(err, "Unable to get MachineConfigPool - worker")
-			return ctrl.Result{}, err
-		}
-
-		// At this time the kata-oc MCP is updated. However the worker MCP might still be in Updating state
-		// We'll need to wait for the worker MCP to complete Updating before deletion
-		r.Log.Info("Wait till worker MCP has updated")
-		if (wMcp.Status.ReadyMachineCount != wMcp.Status.MachineCount) &&
-			mcfgv1.IsMachineConfigPoolConditionTrue(wMcp.Status.Conditions, mcfgv1.MachineConfigPoolUpdating) {
-			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
-		}
-
-		err = r.Client.Delete(context.TODO(), mcp)
-		if err != nil {
-			r.Log.Error(err, "Unable to delete kata-oc MachineConfigPool")
+		r.Log.Info("Get()'ing MachineConfigPool to delete it", "machinePool", "kata-oc")
+		kataOcMcp := &mcfgv1.MachineConfigPool{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: "kata-oc"}, kataOcMcp)
+		if err == nil {
+			r.Log.Info("Deleting MachineConfigPool ", "machinePool", "kata-oc")
+			err = r.Client.Delete(context.TODO(), kataOcMcp)
+			if err != nil {
+				r.Log.Error(err, "Unable to delete kata-oc MachineConfigPool")
+				return ctrl.Result{}, err
+			}
+		} else if k8serrors.IsNotFound(err) {
+			r.Log.Info("MachineConfigPool not found", "machinePool", "kata-oc")
+		} else {
+			r.Log.Error(err, "Unable to get MachineConfigPool ", "machinePool", "kata-oc")
```

Comment:
Most likely, yes, however this is a hard error so it _might_ make sense to log Errors in these cases.  I mean, I kind of doubt it but I'd like to give cases like this proper thought and testing as part of a separate logging clean-up (which is planned and necessary anyway since there's a lot dubious logging in parts of the controller code that wouldn't otherwise be hit by this ongoing clean-up).